docs/001_develop/03_client-capabilities/008_interaction/switch.mdx

```diff
@@ -0,0 +1,149 @@
+---
+title: 'Switch'
+sidebar_label: 'Switch'
+id: client-interaction-switch
+keywords: [interaction, interactive, switch, form]
+tags:
+- interaction
+- interactive
+- switch
+- form
+
+sidebar_position: 8
+---
+import SwitchDemo from '../../../../examples/ui/client-capabilities/interaction/switch.js';
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Switch
+
+A form-connected implementation of a [switch](https://w3c.github.io/aria/#switch) component.
+
+Use cases:
+* Toggling visibility of a UI section
+* Disabling & enabling features
+
+## Example
+
+<SwitchDemo />
+
+<Tabs defaultValue="genesis" values={[{ label: 'Genesis', value: 'genesis', }, { label: 'React', value: 'react', }, { label: 'Angular', value: 'angular', }]}>
+
+<TabItem value="genesis">
+
+Declaration
+```html
+<rapid-switch checked="${checked}"></rapid-switch>
+```
+
+Usage
+```typescript
+@customElement({
+  name: 'my-element',
+  template: html`
+    <rapid-switch>
+      Theme
+      <span slot="checked-message">Dark</span>
+      <span slot="unchecked-message">Light</span>
+    </rapid-switch>
+  `,
+})
+export class MyElement extends GenesisElement {
+}
+```
+
+</TabItem>
+<TabItem value="react">
+
+Declaration
+```html
+<rapid-switch></rapid-switch>
+```
+Usage
+```typescript
+export function MyComponent() {
+  const [switchValue, setSwitchState] = useState(false);
+  const handleSwitchChange = (event) => {
+    setSwitchState(!!event.target.checked);
```

Comment:
is this just resetting the switch state?

Reply:
This is a copy and paste from the checkbox matt did. Im not a react developer, but I think he is manually tracking the state of the checkbox. Feel free to update the code to correct it if you want to

Reply:
https://chatgpt.com/share/672b7fdc-07b4-8009-af8c-73644c6291f1 - checkout this chat gpt link (i'm years out of date with react too)

I created a blank app in create for react and angular to test these things. I understand it's a pain.

If it speeds up the process - mark the section as needs testing so we know to check it later